llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -8800,13 +8815,17 @@ class BoUpSLP::ShuffleCostEstimator : public BaseShuffleAnalysis {
       // Shuffle single vector.
       ExtraCost += GetValueMinBWAffectedCost(V1);
       CommonVF = cast<FixedVectorType>(V1->getType())->getNumElements();
+      if (auto *VecTy = dyn_cast<FixedVectorType>(ScalarTy))
+        CommonVF /= VecTy->getNumElements();
```

Comment:
Add assertion that CommonVF is large enough

Reply:
If `CommonVF` is not large enough, the assert should fail.
```
      assert(all_of(Mask,
                    [=](int Idx) {
                      return Idx < 2 * static_cast<int>(CommonVF);
                    }) &&
             "All elements in mask must be less than 2 * CommonVF.");
```

Reply:
Anyway better to have assertion here

Reply:
Add assertion in `GetVF`.

Reply:
Add assertion in `GetVF`.